ios/Codeforces Watcher/Application/features/problems/ProblemsView.swift

```diff
@@ -0,0 +1,52 @@
+import SwiftUI
+import common
+
+struct ProblemsView: View {
+    
+    var problems: [Problem] = []
+    var noProblemsExplanation = ""
+    
+    var onFilter: () -> Void = {}
+    var onProblem: (String, String) -> Void = { _, _ in }
+    var refreshControl = UIRefreshControl()
+    
+    var body: some View {
+        ZStack {
+            VStack(spacing: 0) {
+                SearchBarView(onFilter: { self.onFilter() })
+                
+                RefreshableScrollView(content: {
+                    if problems.isEmpty {
+                        NoItemsView(imageName: "noItemsProblems", text: noProblemsExplanation)
+                    } else {
+                        ScrollView {
+                            LazyVStack {
+                                ProblemsForEach
+                            }
+                        }
+                    }
+                }, refreshControl: refreshControl)
+                    .background(Palette.white.swiftUIColor)
+                    .cornerRadius(30, corners: [.topLeft, .topRight])
+            }
+        }
+        .background(Palette.accentGrayish.swiftUIColor.edgesIgnoringSafeArea(.top))
+    }
+    
+    @ViewBuilder
+    var ProblemsForEach: some View {
```

Comment:
I'd prefer having `ScrollView` and `LazyVStack` in this component, and name it `ProblemsList`.